### ISSUE TITLE
Deploy bug - socket hang up

### ISSUE THREAD

**Issue by weavermedia** [serverless/serverless#3516]
# This is a Bug Report

## Description

For bug reports:
* What went wrong?
Deploy failed
* What did you expect should have happened?
Deploy succes
* What was the config you used?
Shown in error message below
* What stacktrace or error message from your provider did you see?
```
Error --------------------------------------------------
 
     ServerlessError: ServerlessError: socket hang up
 
     For debugging logs, run again after setting SLS_DEBUG env var.
 
  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
 
     Please report this error. We think it might be a bug.
 
  Your Environment Information -----------------------------
     OS:                 darwin
     Node Version:       6.2.1
     Serverless Version: 1.2.0
```

I'm trying to set environment vars in `serverless.yml` and they weren't accessible through `process.env` in `handler.js`.

I noticed the env vars example in the 'examples' repo contained the following line in the yml file:

`frameworkVersion: ">=1.2.0 <2.0.0"`

so I updated the Serverless CLI from 1.1.0 to 1.2.0 and tried to deploy again. Then I got the error.

**Comment by weavermedia** [comment]
Attempting to deploy again a few minutes later gets this:

```
Serverless Error ---------------------------------------
 
     Stack:arn:aws:cloudformation:us-east-1:970498312278:stack/ffmpeg1-dev/617d4740-287e-11e7-9bc3-50fae9826c99
     is in UPDATE_IN_PROGRESS state and can not be updated.
 
  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
 
  Your Environment Information -----------------------------
     OS:                 darwin
     Node Version:       6.2.1
     Serverless Version: 1.2.0
```

**Comment by mooyoul** [comment]
@weavermedia [AWS seems currently down](https://twitter.com/search?f=tweets&q=Aws&src=typd), 

BTW amazon.com also seems down.
<img width="1323" alt="2017-04-25 3 39 36" src="https://cloud.githubusercontent.com/assets/2101743/25371962/53d0e682-29ce-11e7-9e49-b7550d8c9336.png">

**Comment by pmuens** [comment]
@weavermedia thanks for reporting. The second error occurs because you're trying to update the stack while it's in a rollback state.

Could you run `serverless deploy --verbose again` and paste the output here so that we can see what's going on at AWS end. Thanks!

**Comment by MatissJanis** [comment]
I'm running into the exact same issue.

```
$ sls deploy -v
Serverless: Packaging service...
Serverless: Uploading CloudFormation file to S3...
Serverless: Uploading function .zip files to S3...
 
  Serverless Error ---------------------------------------
 
     Your socket connection to the server was not read from
     or written to within the timeout period. Idle connections
     will be closed.
 
  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Forums:        forum.serverless.com
     Chat:          gitter.im/serverless/serverless
 
  Your Environment Information -----------------------------
     OS:                 darwin
     Node Version:       7.8.0
     Serverless Version: 1.11.0
```

**Comment by pmuens** [comment]
Thanks again for reporting. Closing this one for now since we're not able to reproduce it and it seems like it's smth. on AWS end.

Feel free to re-open if this bug persists.

**Comment by seawatts** [comment]
I'm actually having this same error now.

**Comment by SoftMaker** [comment]
Yesterday everything was OK, today:

`Error: socket hang up
    at createHangUpError (_http_client.js:313:15)
    at TLSSocket.socketOnEnd (_http_client.js:416:23)
    at TLSSocket.emit (events.js:187:15)
    at TLSSocket.EventEmitter.emit (domain.js:442:20)
    at endReadableNT (_stream_readable.js:1081:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)`

**Comment by vladimirdjuricic** [comment]
I encountered the same thing with node version "10.7.0". With node version "10.6.0" works just fine. Hope this helps.

**Comment by eahefnawy** [comment]
@SoftMaker We just released a new v1.29.0 version that includes the fix for this. Could you try upgrading serverless?

**Comment by SoftMaker** [comment]
@eahefnawy After upgrading to v1.29.0 everything is OK. Thanks!
